### PR TITLE
[5920] Close content type form after clicking on "Save & Close"

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5920

### CODE REVIEW

**Review by rart — CHANGES_REQUESTED** [review]
On the "no template dialog"...
-  Update title to "Missing Template"
- Update the body: "Confirm a template is not required for this content type. Failing to assign a template would cause rendering issues on templated apps."
- Update the button choices to: 
  - **Template not required, save & close** — Clicking would _check_ the "No template required" option, save & close
  - **Create a template** — Open the template creation experience. If completed, use that as the template, save & close. If cancelled, for simplicity, dismiss the dialog and remain in the editor.
  - **Choose existing template** — Open the template picking experience. Completed/cancelled same as "create"
  - **Stay & continue editing**

Use a dialog layout like the "Confirm Dialog" with centered copy and full width stacked buttons.

**Review by rart — CHANGES_REQUESTED** [review]
Refactor the ConfirmDialog to internally use a new `AlertDialog`. The alert dialog would be basically what the ConfirmDialog uses underneath. AlertDialog receives prop `buttons: React.ReactNode`. Get rid of the usage of makeStyles and put everything in place with sx. I think we can even remove the `classes` prop from confirm for now.

Remove the `NewTemplateDialog` components, this is too specific and probably soon to be discontinued. Simply render the new `AlertDialog` with the right props on the type editor.

File: ui/app/src/env/i18n-legacy.ts [hunk 1]
```diff
@@ -435,6 +435,39 @@ export const contentTypesMessages = defineMessages({
...
+  },
+  missingTemplateBody: {
+    id: 'contentType.missingTemplateBody',
+    defaultMessage:
+      'Confirm a template is not required for this content type. Failing to assign a template would cause rendering issues on templated apps.'
+  },
+  templateNotRequiredSave: {
+    id: 'contentType.templateNotRequiredSave',
+    defaultMessage: 'Template not required, save'
+  },
```
Comment (rart) [inline ui/app/src/env/i18n-legacy.ts:451]: Is this in use?

File: ui/app/src/components/AlertDialog/AlertDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    open,
+    disableBackdropClick,
+    disableEnforceFocus,
+    hideBackdrop,
+    maxWidth,
+    disableEscapeKeyDown,
+    onClose,
+    ...rest
+  } = props;
+  const sx = getStyles();
```
Comment (rart) [inline ui/app/src/components/AlertDialog/AlertDialog.tsx]: Move the styles in place

File: ui/app/src/components/AlertDialog/AlertDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    disableBackdropClick: disableBackdropClick,
+    disableEscapeKeyDown: disableEscapeKeyDown
+  });
+
+  return (
+    <Dialog
+      open={open}
+      onClose={onCloseHandler}
+      aria-labelledby="confirmDialogTitle"
+      aria-describedby="confirmDialogBody"
```
Comment (rart) [inline ui/app/src/components/AlertDialog/AlertDialog.tsx]: 👆

File: ui/app/src/components/AlertDialog/AlertDialogContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import { nnou } from '../../utils/object';
+
+export function AlertDialogContainer(props: AlertDialogContainerProps) {
+  const { onClosed, body, title, children, imageUrl = questionGraphicUrl, buttons } = props;
+  const sx = getStyles();
+
+  useUnmount(onClosed);
+  return (
+    <>
+      <DialogContent id="confirmDialogBody" sx={sx.dialogBody}>
```
Comment (rart) [inline ui/app/src/components/AlertDialog/AlertDialogContainer.tsx]: 👆

File: ui/app/src/components/AlertDialog/AlertDialogContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import Typography from '@mui/material/Typography';
+import DialogContentText from '@mui/material/DialogContentText';
+import { getStyles } from './styles';
+import Box from '@mui/material/Box';
+import { DialogFooter } from '../DialogFooter';
+import { nnou } from '../../utils/object';
+
+export function AlertDialogContainer(props: AlertDialogContainerProps) {
+  const { onClosed, body, title, children, imageUrl = questionGraphicUrl, buttons } = props;
+  const sx = getStyles();
```
Comment (rart) [inline ui/app/src/components/AlertDialog/AlertDialogContainer.tsx]: Move styles in place

File: ui/app/src/components/AlertDialog/styles.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
+/*
```
Comment (rart) [inline ui/app/src/components/AlertDialog/styles.ts]: Discard file

File: ui/app/src/components/AlertDialog/styles.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { FullSxRecord, PartialSxRecord } from '../../models';
+
+export type AlertDialogClassKey = 'dialog' | 'dialogImage' | 'dialogBody' | 'dialogTitle' | 'dialogFooter';
```
Comment (rart) [inline ui/app/src/components/AlertDialog/styles.ts]: Update these keys.
Need to declare the `sxs?: PartialSxRecord<'root' | 'image' | 'body' | 'title' | 'footer'>` prop on the dialog props

File: ui/app/src/components/AlertDialog/AlertDialogContainer.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+  useUnmount(onClosed);
+  return (
+    <>
+      <DialogContent id="confirmDialogBody" sx={sx.dialogBody}>
+        <Box component="img" src={imageUrl} alt="" sx={sx.dialogImage} />
```
Comment (rart) [inline ui/app/src/components/AlertDialog/AlertDialogContainer.tsx]: Wrap image in conditional like title and body so we could have the dialog without image too

File: ui/app/src/components/AlertDialog/AlertDialogContainer.tsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+          </Typography>
+        )}
+        {body && (
+          <DialogContentText color="textPrimary" variant="body2">
+            {body}
+          </DialogContentText>
+        )}
+        {children}
+      </DialogContent>
+      <DialogFooter sx={sx.dialogFooter}>{nnou(buttons) && buttons}</DialogFooter>
```
Comment (rart) [inline ui/app/src/components/AlertDialog/AlertDialogContainer.tsx]:
```suggestion
      {nnou(buttons) && <DialogFooter sx={sxs?.footer}>{buttons}</DialogFooter>}
```

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/env/i18n-legacy.ts (hunk 1 above)
Comment (jvega190) [reply ui/app/src/env/i18n-legacy.ts:451]: yeah that's one of the options when saving the content-type editor. (save, save & close, save & minimize)

File: ui/app/src/components/AlertDialog/AlertDialog.tsx (hunk 2 above)
Comment (rart) [reply ui/app/src/components/AlertDialog/AlertDialog.tsx]: This wasn't done.

File: ui/app/src/components/AlertDialog/AlertDialogContainer.tsx (hunk 5 above)
Comment (rart) [reply ui/app/src/components/AlertDialog/AlertDialogContainer.tsx]: This wasn't done.

**Review by rart — APPROVED** [review]

**Pull request merged**